lms/djangoapps/grades/rest_api/v1/gradebook_views.py

```diff
@@ -515,22 +525,27 @@ def get(self, request, course_key):
             serializer = StudentGradebookEntrySerializer(entry)
             return Response(serializer.data)
         else:
-            filter_kwargs = {}
-            related_models = []
-            if request.GET.get('username_contains'):
-                filter_kwargs['user__username__icontains'] = request.GET.get('username_contains')
-                related_models.append('user')
+            q_objects = []
+            if request.GET.get('user_contains'):
+                search_term = request.GET.get('user_contains')
+
+                q_objects.append(
+                    Q(user__username__icontains=search_term) |
+                    Q(programcourseenrollment__program_enrollment__external_user_key__icontains=search_term) |
```

Comment:
I wonder if this should be `istartswith`, especially if the user key is likely to be a number.

Reply:
I thought about this a bit and I think we should probably not make the platform opinionated about the sort of semantics of this. I could understand if we wanted to change all these fields to `istartswith`, though, but I think educators may be frustrated with differential behavior between our usernames and their keys (which may be, opaquely to us, usernames in the educator's mind).